pipeline/ais-annotator/src/main/scala/AISAnnotator.scala

```diff
@@ -211,9 +219,25 @@ object AISAnnotator extends LazyLogging {
       }
 
       val annotated = annotateAllMessages(includedMMSIs, inputData, annotations)
-      val annotatedToString = annotated.map(json => compact(render(json)))
 
-      annotatedToString.saveAsTextFile(outputFilePath)
+      val taggedByShard = annotated.map { json => 
+        val dateTime = Instant.parse(json.getString("timestamp").replace(" UTC", "Z").replace(" ", "T")).toDateTime(UTC)
+        val text = compact(render(json))
+        val minuteKey = (dateTime.getMinuteOfDay():Int) / 10
+        ((dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth(), minuteKey), text)
+      }
+      .groupByKey
+      .map {
+        case ((year, month, day, shard_no), groupedByMinute) => {
+          val shardName = "$year%04d-$month%02d-$day%02d"
```

Comment:
The string literal needs to be prefixed with 's' to get string interpolation. E.g.

s"$year%04d-$month%02d-$day%02d"